File: solutions/899.binary-gap/binary-gap.java
class Solution {
    public int binaryGap(int N) {
        int res = 0, curRes = -32;
        while(N > 0) {
            if((N & 1) == 1) {
                res = Math.max(res, curRes);
                curRes = 0;
            }
            N >>>= 1;
            curRes++;
        }
        return res;
    }
}

File: solutions/899.binary-gap/binary-gap.cpp
class Solution {
public:
    int binaryGap(int N) {
        int j = -1, res = 0;
        for(int i = 0; i < 31; i++) {
            if(N & (1 << i)) {
                if(j >= 0) 
                    res = max(res, i - j);
                j = i;
            }
        }
        return res;
    }
};